component.go

```diff
@@ -99,6 +105,107 @@ func (c *Component) Render(ctx context.Context) (template.HTML, error) {
 	return c.renderNamed(ctx, filepath.Base(c.templates[0]), c.templates, c.templateData)
 }
 
+func (c *Component) RenderWithRequest(ctx context.Context, r *http.Request) (template.HTML, error) {
+	// Check for circular references
+	if ctx.Value(c) != nil {
+		return "", errors.New("circular reference detected in partials")
+	}
+
+	// Add current component to context
+	ctx = context.WithValue(ctx, c, true)
+
+	// Check if this is an HTMX request
+	hxTarget := r.Header.Get("HX-Target")
+
+	// Handle partial rendering if this is an HTMX request
+	if RenderPartial(r) {
+		if hxTarget == "" {
```

Comment:
I think [hxSelect](https://htmx.org/attributes/hx-select/) is the correct header to use for partial rendering. If you had a full template, hx-select picks the element to swap into the target.

Reply:
"That's interesting! Though I think that hx-select is meant to extract a part of the full response, not to control what the backend returns. 

I could see adding a small HTMX plugin like `hx-partial` to have the backend only return the specific part we need. Personally, I’d prefer to avoid having the backend render everything, only to then filter out most of it.

Reply:
I feel like we're talking about two different things. We need a way to talk about an endpoint designed to return a html partial (equivalent to a JSON document) vs a full page for the browser. I don't think Component describes either of these things very well.